### ISSUE TITLE
Variable `homebrew_uninstalled_packages` is undefined

### ISSUE THREAD

**Issue by MikeSouza** [geerlingguy/ansible-role-homebrew#66]
<img width="725" alt="ansible-role-homebrew_uninstalled_packages" src="https://cloud.githubusercontent.com/assets/1017268/24099593/8e6dbde6-0d46-11e7-9ff2-6278fdb074aa.png">

I think this is a bug, but I'm not sure if it's something with my particular ansible configuration (default) or what.  Based on commits  50c72f2daf64cf1c5bada40b65bb57c0eeb197ae and b5bb8f5166e810329139cc489717011c6465e616, I can't tell whether it's just me getting this error or not.

In my existing and new playbooks dependent on this role, I **MUST** define the variable `homebrew_uninstalled_packages: []` else I get the error above.

Anyways, the simple fix is to uncomment the `homebrew_uninstalled_packages` variable in `defaults/main.yml`.  I'm just unsure if there was a reason for commenting it out in the first place.

Let me know if you need any further details.  Thanks.

**Comment by elliotweiser** [comment]
IIRC, Ansible 2.0+ dislikes undefined variables, which seems to obviate the `is defined` check. My guess is the "best" (or at least easiest) fix is to uncomment the offending variable in `defaults/main.yml`, thus defining it. Sane default; easily overridden.

**Comment by geerlingguy** [comment]
There's another way of defining that `when` condition, though... I can't recall right now and can't look up in my recent commits to other roles, but you basically have to do two checks to see if it's defined and has a value—I think. It might be safer/more sane, otherwise I'm okay with just uncommenting that default var if it doesn't cause any other issues.

**Comment by MikeSouza** [comment]
I've determined the problem.  The `with_items: {{ homebrew_uninstalled_packages }}` is evaluated before the `when` clause.  This is intended behavior (See ansible/ansible#13950). 

This can also be fixed by changing it to:  `with_items: {{ homebrew_uninstalled_packages|default([]) }}`

However, IMO uncommenting the line in `defaults/main.yml` and removing the `when` clause altogether is the better solution for the following reasons:

- From the POV of someone new who's using this role, it's nice to be able to just look in the `defaults/main.yml` to see all the available variables and their defaults, especially when every other variable is defined and defaulted in this file.
- The task used to install packages with the `homebrew_installed_packages` variable is practically the same as this task, just for installed packages, and I don't see a good reason why this task _should_ be different.

Anyways, I've submitted PR #67 with a fix for this.
